pkg/reader/usdc_reader.go

```diff
@@ -0,0 +1,308 @@
+package reader
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	sel "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+
+	"github.com/smartcontractkit/chainlink-ccip/execute/exectypes"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
+)
+
+type MessageHash []byte
+
+type USDCMessageReader interface {
+	MessageHashes(ctx context.Context,
+		source, dest cciptypes.ChainSelector,
+		tokens map[exectypes.MessageTokenID]cciptypes.RampTokenAmount,
+	) (map[exectypes.MessageTokenID]MessageHash, error)
+}
+
+const (
+	CCTPMessageVersion = uint32(0)
+)
+
+// TODO, this should be fetched from USDC Token Pool and cached
```

Comment:
Separate PR, this values are fixed, but we should fetch them from chain instead of hardcoding to rely on a single source of the truth